ingenialink/ethercat/network.py

```diff
@@ -28,6 +30,69 @@
 
 logger = ingenialogger.get_logger(__name__)
 
+# Holds a reference to the Ethercat network by the time it is created -> handle no-GIL cases
+ETHERCAT_NETWORK_REFERENCES: dict[float, "EthercatNetwork"] = {}
+
+
+def set_network_reference(network: "EthercatNetwork") -> float:
+    """Adds a reference to an EtherCAT network.
+
+    Args:
+        network: network.
+
+    Returns:
+        network creation time.
+    """
+    creation_time_s = time.time()
+    ETHERCAT_NETWORK_REFERENCES[creation_time_s] = network
+    return creation_time_s
+
+
+def release_network_reference(creation_time_s: float) -> None:
+    """Releases a network reference.
+
+    Args:
+        creation_time_s: network creation time.
+    """
+    ETHERCAT_NETWORK_REFERENCES.pop(creation_time_s)
+
+
+@dataclass
+class GilReleaseConfig:
+    """Configuration of pysoem functions that have GIL release control."""
+
+    always_release: Optional[bool] = None
```

Comment:
Alternatively this could be a classmethod
````
GilReleaseConfig:
...
@classmethod
def always(cls):
return cls(config_init=True, ....=True,...)


config = GilReleaseConfig.always()
````